tests/src/main/java/org/hibernate/beanvalidation/tck/util/TestUtil.java

```diff
@@ -167,16 +167,8 @@ public static void assertCorrectConstraintTypes(Set<? extends ConstraintViolatio
 		);
 
 		for ( String propertyPath : propertyPaths ) {
-			Path expectedPath = PathImpl.createPathFromString( propertyPath );
```

Comment:
Hum, it seems risky to solely base the assertion on the string representation, given the spec doesn't mandate any particular one. Perhaps assertCorrectPropertyPaths() should take a more object-orientated representation of expected paths (we still can leave the current one for the most common case of plain non-indexed, non-keyed props). If you agree, can you open a follow-up BVTCK issue for it?